Geometry/HcalTowerAlgo/src/CaloTowerGeometry.cc

```diff
@@ -5,51 +5,48 @@ typedef CaloCellGeometry::CCGFloat CCGFloat ;
 typedef CaloCellGeometry::Pt3D     Pt3D     ;
 typedef CaloCellGeometry::Pt3DVec  Pt3DVec  ;
 
-CaloTowerGeometry::CaloTowerGeometry() :
-   m_cellVec ( k_NumberOfCellsForCorners ) 
+CaloTowerGeometry::CaloTowerGeometry(const CaloTowerTopology *cttopo_) :
+  cttopo(cttopo_), 
+  k_NumberOfCellsForCorners(cttopo->sizeForDenseIndexing()), 
+  k_NumberOfShapes(cttopo->lastHFRing()), 
+  m_cellVec ( k_NumberOfCellsForCorners ) 
 {
 }
   
 
-CaloTowerGeometry::~CaloTowerGeometry()
-{
-}
-
-
-unsigned int
-CaloTowerGeometry::alignmentTransformIndexLocal( const DetId& id )
-{
-   const CaloGenericDetId gid ( id ) ;
-
-   assert( gid.isCaloTower() ) ;
-
-   const CaloTowerDetId cid ( id ) ;
+CaloTowerGeometry::~CaloTowerGeometry() { }
 
-   const unsigned int iea ( cid.ietaAbs() ) ;
 
-   const unsigned int ip ( ( cid.iphi() - 1 )/4 ) ;
-
-   const int izoff ( ( cid.zside() + 1 )/2 ) ;
-
-   const unsigned int offset ( izoff*3*18) ;
+unsigned int 
+CaloTowerGeometry::alignmentTransformIndexLocal( const DetId& id ) {
+  
+  const CaloGenericDetId gid ( id ) ;
+  assert( gid.isCaloTower() ) ;
+
+  const CaloTowerDetId cid ( id ) ;
+  const int iea ( cid.ietaAbs() ) ;
+  const unsigned int ip ( ( cid.iphi() - 1 )/4 ) ;
+  const int izoff ( ( cid.zside() + 1 )/2 ) ;
+  const unsigned int offset ( izoff*3*18) ;
+  
+  assert(0);
```

Comment:
@kpedro88 - usually, when a function ends with such an assert is because the programmer wants to catch cases where he didn't cover some possibility. In this case you'll not return what you expect to return after the assert.